free-3.md

```diff
@@ -4,51 +4,71 @@
+
 
 ### Why a language?
 _Why is a DSL appropriate for your user(s)? How does it address the need?_
 
+A DSL is appropriate because we want the users to give input on what kind of sheet music is generated for sight reading. A language can express a lot, and this kind of detail can create sheet music tailored to individual students and their needs. It addresses the problem of finding the right kind of music to sight read.
 
 ### Why you?
 _What excites you about this idea? How did you come up with it?_
 
+From my first project description, I explained how I am involved in music. However, I cannot sight read as well as some others. In the world of classical music, many professionals are able to sight read fairly difficult material, whereas I take time in reading the music. I sometimes blame my poor eyesight and lack of new glasses, but the main cause of my relatively poor sight reading skills comes from the lack of practice and material.
+
+I really hope people can get good sight reading material easily that is tailored to them and free. I don't want others following my footsteps in not practicing sight reading because they didn't have enough accessible material.
+
 
 ### Domain
 _Describe the project's domain in five words._
-
+Sight Reading Music Generation
 
 ### Interface (syntax)
 _How might the user interact with the language? What does programming look 
 like? Why is this the right way to interact with the problem domain?_ 
 
+A user would probably write some form of code, specifying the constraints of the music generation. Programming would probably look like code in Python. I believe this is a good way to interact with the problem domain because the only thing we need to do is specify constraints in the music generation. Writing statements is perhaps the easiest.
 
 ### Operation (semantics)
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of errors might occur, and how might they be communicated to
 the user?_
 
+Some random sheet music should be formed based on the constraints given. The program interacts with the user by giving the sheet music back. I think errors should occur if the constraints/specifications given to the program contradict each other. This should be resolved as a message or perhaps as a blank sheet of music.
 
 ### Expressiveness
 _What should be easy to do in this language? What should be possible, but
 difficult? What should be impossible or very difficult?_
 
+It should be easy to specify what kind of music you want. This includes the meter, time, rhythms, and possible harmonies. It could be possible for someone to write specific notes into the music, but I would limit the number of notes someone can write in. If someone wants to write all the notes, they could simply use some existing software. 
```

Comment:
I'd be very curious to see what randomly generated music would sound like. Is it meant to have melodies or could it just generate a random sequence of notes with no real relationship between the notes? It seems like setting certain parameters like meter, time and rhythm is within the scope of a CS111 project, but having harmonies would be exponentially harder to program. Could be wrong, but that's just my two cents.